File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums_set = set(nums)
        maxSequence = 0
        for n in nums_set:
            if n - 1 in nums_set:
                continue
            count = 1
            while (n + count)  in nums_set:  
                count += 1
            maxSequence = max(maxSequence, count)
        return maxSequence          

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int goodNodes(TreeNode root) {
        if(root == null)
          return 0;
        return helper(root, root.val);  
    }

    private int helper(TreeNode node, int max)
    {
        if(node == null)
        {
            return 0;
        }
        int count=0;

        if(node.val >= max)
        {
          max=node.val;
          count++;
        }
       int leftCount = helper(node.left, max);
       int rightCount = helper(node.right, max);
        return count+leftCount+rightCount;
        
    }
}
